src/Chainweb/Payload.hs

```diff
@@ -774,18 +775,29 @@ instance IsCasValue PayloadWithOutputs where
     {-# INLINE casKey #-}
 
 payloadWithOutputs
```

Comment:
Add a haddock comment that makes the preconditions of the function explicit: Transaction and output vectors are of same length and have the tx and the matching outputs are at the same index.